src/core/tiledmesh/qgstiledmeshprovidermetadata.h

```diff
@@ -0,0 +1,51 @@
+/***************************************************************************
+  qgstiledmeshprovidermetadata.h
+  --------------------------------------
+  Date                 : June 2023
+  Copyright            : (C) 2023 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSTILEDMESHPROVIDERMETADATA_H
+#define QGSTILEDMESHPROVIDERMETADATA_H
+
+
+#include "qgsprovidermetadata.h"
+
+///@cond PRIVATE
+#define SIP_NO_FILE
+
+/**
+ * This metadata class is responsible for generic tiled mesh handling.
+ *
+ * Specific tiled mesh data provider classes will also implement their own
+ * QgsProviderMetadata.
```

Comment:
Even though it's a private, maybe add a \since for those venturing in this area in the future?